internal/pool/pool.go

```diff
@@ -121,14 +127,17 @@ func (p *ConnPool) checkMinIdleConns() {
 	for p.poolSize < p.opt.PoolSize && p.idleConnsLen < p.opt.MinIdleConns {
```

Comment:
You could also add `&& !p.closed()` just in case.

Reply:
Or even better, check `p.closed()` in `popIdle`?

Reply:
`checkMinIdleConns` is executed after `connsMu.Lock()`, it protects `poolSize` and `idleConnsLen`, we only need to ensure their correctness in the goroutine (func).

```go
if p.opt.MinIdleConns == 0 || p.closed() {
	return
}
```
We can add a `p.closed()` judgment here, but its function is only to not start the goroutine.

Reply:
Perhaps adding something like this to `Pool.Get`:

```
		p.connsMu.Lock()
                if p.Closed() {
                    return nil, ErrClosed
                }
		cn := p.popIdle()
		p.connsMu.Unlock()
```

Reply:
Although `p.Get()` can still get the connection after `Close()`, it will not cause the connection pool to leak.

Reply:
We have added detection (`if p.closed()`) in `p.newConn()`, `p.Get()` will not add unmanaged new connections to the closed connection pool.